src/Makefile

```diff
@@ -328,19 +326,22 @@ LIBCARTESI_OBJS:= \
 	base64.o \
 	interpret.o \
 	virtual-machine.o \
-	machine-c-api.o \
 	uarch-machine.o \
 	uarch-step.o \
-	uarch-interpret.o
+	uarch-interpret.o \
+	machine-c-api.o
+
+CARTESI_CLUA_OBJS:= \
+	clua.o \
+	clua-i-virtual-machine.o \
+	clua-htif.o \
+	clua-machine-util.o
 
 LUACARTESI_OBJS:= \
-	sha3.o \
 	clua-cartesi.o \
-	clua-i-virtual-machine.o \
 	clua-machine.o \
-	clua-htif.o \
-	clua-machine-util.o \
-	clua.o
+	$(CARTESI_CLUA_OBJS) \
+	$(LIBCARTESI_OBJS)
```

Comment:
Shouldn´t LUACARTESI simply use LIBCARTESI?

Reply:
It's almost the same thing, `libcartesi.a` is just a "zip" of all `.o` objects. By using objects directly instead of `.a` the build can skip the `ar rcs` step, so it's more error prone in case the user forget to ser `AR` variable when cross compiling and he wants just to build `.so` files.

Reply:
I am reworking to *make use* of the static libraries.